### ISSUE TITLE
symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno

### ISSUE THREAD

**Issue by AdamIsrael** [ubuntu/microk8s#2476]
This just started happening today, after an auto-refresh to v1.20.9. It persists even after a `snap restart microk8s`.

```bash
$ microk8s.status
microk8s is running
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
high-availability: no
  datastore master nodes: none
  datastore standby nodes: none
addons:
  enabled:
    dns                  # CoreDNS
    ha-cluster           # Configure high availability on the current node
    storage              # Storage class; allocates storage from host directory
  disabled:
    [...]
```

[inspection-report-20210726_165803.tar.gz](https://github.com/ubuntu/microk8s/files/6881200/inspection-report-20210726_165803.tar.gz)

**Comment by polyedre** [comment]
I could reproduce the bug on a RHEL 8 with:

```sh
sudo snap install microk8s --classic --channel=1.20/stable
microk8s disable ha-cluster
```

**Comment by balchua** [comment]
Looking at the logs, the apiserver seems to be up.  I think you can do a `kubectl get no` to see if u can access the cluster.
But what is strange is that the calico node isn't ready.  If you delete calico-node does it come back to life?

**Comment by balchua** [comment]
@mathieubordere. Do you have an idea why `dqlite` shows these error?
```
/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
```
Although the cluster seems to be up.

**Comment by luca-nardelli** [comment]
Not sure if it helps, but updating to 1.21 fixed the issue for me

**Comment by MathieuBordere** [comment]
@balchua @ktsakalozos mentioned this issue in the dqlite channel on mattermost  on 13th of July and seems to be related to how you are building the snap if I'm not mistaken, can you check with him please?

**Comment by balchua** [comment]
It looks like the fix on this is already merged.
https://github.com/ubuntu/microk8s/pull/2428
I wonder if this is already released.

**Comment by AdamIsrael** [comment]
> I wonder if this is already released.

Not yet. Hopefully a new release will be forthcoming.

**Comment by chiefnoah** [comment]
This is effecting our production cluster due to `snap`'s auto update. What is the timeframe for the fix to be released?

**Comment by receperdogan** [comment]
> This just started happening today, after an auto-refresh to v1.20.9. It persists even after a `snap restart microk8s`.
> 
> ```shell
> $ microk8s.status
> microk8s is running
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> /snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno
> high-availability: no
>   datastore master nodes: none
>   datastore standby nodes: none
> addons:
>   enabled:
>     dns                  # CoreDNS
>     ha-cluster           # Configure high availability on the current node
>     storage              # Storage class; allocates storage from host directory
>   disabled:
>     [...]
> ```
 

same here..
`microk8s inspect`  works properly
but `microk8s status` gives same output

**Comment by lfdominguez** [comment]
my cluster crash thanks to snap autorefresh.... whyyy this kind of thing doesnt have a switch
`/snap/microk8s/2338/bin/dqlite: symbol lookup error: /snap/microk8s/2338/bin/dqlite: undefined symbol: sqlite3_system_errno`

**Comment by StanSocialCloud** [comment]
Where can I track upcoming patches in 1.20, I reverted most of the machines back to 1.20.7, but some are still broken due to hard snap removals during debugging(can't revert), so it would be best if everything is updated to the upcoming patch with the merged fixes.

**Comment by ktsakalozos** [comment]
New snap revisions (2361 and 2362) should be on 1.20/edge, beta and candidate. This revision is now going through testing to be released on the 1.20/stable channel.

As @balchua mentioned, this issue was patched a few weeks ago in all tracks (1.19, 1.20, 1.21 and latest). Our CI did not pick up the change on the 1.20 branch. I cannot offer more insight as to why this is the case, I need to do some  investigation.

Apologies for the inconvenience.

**Comment by geowalrus4gh** [comment]
Getting the same issue here. will this be a problem to add a node to the cluster? Am not able to join a new node to the cluster.

**Comment by ktsakalozos** [comment]
> Getting the same issue here.

@geowalrus4gh, did the node refresh to `2361` or `2362`? What do you see with `snap info microk8s` ?

**Comment by geowalrus4gh** [comment]
snap-id:      EaXqgt1lyCaxKaQCU349mlodBkDCXRcg
tracking:     1.20/stable
refresh-date: 7 days ago, at 11:37 UTC
channels:
  1.21/stable:      v1.21.3         2021-07-27 (2346) 191MB classic
  1.21/candidate:   v1.21.3         2021-07-20 (2346) 191MB classic
  1.21/beta:        v1.21.3         2021-07-20 (2346) 191MB classic
  1.21/edge:        v1.21.3         2021-07-23 (2354) 191MB classic
  latest/stable:    v1.21.3         2021-07-28 (2346) 191MB classic
  latest/candidate: v1.21.3         2021-07-17 (2337) 193MB classic
  latest/beta:      v1.21.3         2021-07-17 (2337) 193MB classic
  latest/edge:      v1.21.3         2021-07-28 (2359) 193MB classic
  dqlite/stable:    –
  dqlite/candidate: –
  dqlite/beta:      –
  dqlite/edge:      v1.16.2         2019-11-07 (1038) 189MB classic
  1.22/stable:      –
  1.22/candidate:   v1.22.0-rc.0    2021-07-21 (2349) 197MB classic
  1.22/beta:        v1.22.0-beta.2  2021-07-17 (2348) 197MB classic
  1.22/edge:        v1.22.0-alpha.3 2021-06-18 (2284) 196MB classic
  1.20/stable:      v1.20.9         2021-08-01 (2361) 221MB classic
  1.20/candidate:   v1.20.9         2021-08-01 (2361) 221MB classic
  1.20/beta:        v1.20.9         2021-08-01 (2361) 221MB classic
  1.20/edge:        v1.20.9         2021-08-01 (2361) 221MB classic
  1.19/stable:      v1.19.13        2021-07-26 (2339) 216MB classic
  1.19/candidate:   v1.19.13        2021-07-19 (2339) 216MB classic
  1.19/beta:        v1.19.13        2021-07-19 (2339) 216MB classic
  1.19/edge:        v1.19.13        2021-07-16 (2339) 216MB classic
  1.18/stable:      v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/candidate:   v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/beta:        v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/edge:        v1.18.20        2021-06-16 (2271) 198MB classic
  1.17/stable:      v1.17.17        2021-01-15 (1916) 177MB classic
  1.17/candidate:   v1.17.17        2021-01-14 (1916) 177MB classic
  1.17/beta:        v1.17.17        2021-01-14 (1916) 177MB classic
  1.17/edge:        v1.17.17        2021-01-13 (1916) 177MB classic
  1.16/stable:      v1.16.15        2020-09-12 (1671) 179MB classic
  1.16/candidate:   v1.16.15        2020-09-04 (1671) 179MB classic
  1.16/beta:        v1.16.15        2020-09-04 (1671) 179MB classic
  1.16/edge:        v1.16.15        2020-09-02 (1671) 179MB classic
  1.15/stable:      v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/candidate:   v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/beta:        v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/edge:        v1.15.11        2020-03-26 (1301) 171MB classic
  1.14/stable:      v1.14.10        2020-01-06 (1120) 217MB classic
  1.14/candidate:   ↑
  1.14/beta:        ↑
  1.14/edge:        v1.14.10        2020-03-26 (1303) 217MB classic
  1.13/stable:      v1.13.6         2019-06-06  (581) 237MB classic
  1.13/candidate:   ↑
  1.13/beta:        ↑
  1.13/edge:        ↑
  1.12/stable:      v1.12.9         2019-06-06  (612) 259MB classic
  1.12/candidate:   ↑
  1.12/beta:        ↑
  1.12/edge:        ↑
  1.11/stable:      v1.11.10        2019-05-10  (557) 258MB classic
  1.11/candidate:   ↑
  1.11/beta:        ↑
  1.11/edge:        ↑
  1.10/stable:      v1.10.13        2019-04-22  (546) 222MB classic
  1.10/candidate:   ↑
  1.10/beta:        ↑
  1.10/edge:        ↑
installed:          v1.20.9                    (2338) 221MB classic

**Comment by ktsakalozos** [comment]
Can you do a `snap refresh microk8s` ?

**Comment by geowalrus4gh** [comment]
> Can you do a `snap refresh microk8s` ?

Cool. That worked. Saved my day. thanks @ktsakalozos 

This is my current status. No sqlite error now and added new node too. 

tracking:     1.20/stable
refresh-date: today at 08:35 UTC
channels:
  1.21/stable:      v1.21.3         2021-07-27 (2346) 191MB classic
  1.21/candidate:   v1.21.3         2021-07-20 (2346) 191MB classic
  1.21/beta:        v1.21.3         2021-07-20 (2346) 191MB classic
  1.21/edge:        v1.21.3         2021-07-23 (2354) 191MB classic
  latest/stable:    v1.21.3         2021-07-28 (2346) 191MB classic
  latest/candidate: v1.21.3         2021-07-17 (2337) 193MB classic
  latest/beta:      v1.21.3         2021-07-17 (2337) 193MB classic
  latest/edge:      v1.21.3         2021-07-28 (2359) 193MB classic
  dqlite/stable:    –
  dqlite/candidate: –
  dqlite/beta:      –
  dqlite/edge:      v1.16.2         2019-11-07 (1038) 189MB classic
  1.22/stable:      –
  1.22/candidate:   v1.22.0-rc.0    2021-07-21 (2349) 197MB classic
  1.22/beta:        v1.22.0-beta.2  2021-07-17 (2348) 197MB classic
  1.22/edge:        v1.22.0-alpha.3 2021-06-18 (2284) 196MB classic
  1.20/stable:      v1.20.9         2021-08-01 (2361) 221MB classic
  1.20/candidate:   v1.20.9         2021-08-01 (2361) 221MB classic
  1.20/beta:        v1.20.9         2021-08-01 (2361) 221MB classic
  1.20/edge:        v1.20.9         2021-08-01 (2361) 221MB classic
  1.19/stable:      v1.19.13        2021-07-26 (2339) 216MB classic
  1.19/candidate:   v1.19.13        2021-07-19 (2339) 216MB classic
  1.19/beta:        v1.19.13        2021-07-19 (2339) 216MB classic
  1.19/edge:        v1.19.13        2021-07-16 (2339) 216MB classic
  1.18/stable:      v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/candidate:   v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/beta:        v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/edge:        v1.18.20        2021-06-16 (2271) 198MB classic
  1.17/stable:      v1.17.17        2021-01-15 (1916) 177MB classic
  1.17/candidate:   v1.17.17        2021-01-14 (1916) 177MB classic
  1.17/beta:        v1.17.17        2021-01-14 (1916) 177MB classic
  1.17/edge:        v1.17.17        2021-01-13 (1916) 177MB classic
  1.16/stable:      v1.16.15        2020-09-12 (1671) 179MB classic
  1.16/candidate:   v1.16.15        2020-09-04 (1671) 179MB classic
  1.16/beta:        v1.16.15        2020-09-04 (1671) 179MB classic
  1.16/edge:        v1.16.15        2020-09-02 (1671) 179MB classic
  1.15/stable:      v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/candidate:   v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/beta:        v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/edge:        v1.15.11        2020-03-26 (1301) 171MB classic
  1.14/stable:      v1.14.10        2020-01-06 (1120) 217MB classic
  1.14/candidate:   ↑
  1.14/beta:        ↑
  1.14/edge:        v1.14.10        2020-03-26 (1303) 217MB classic
  1.13/stable:      v1.13.6         2019-06-06  (581) 237MB classic
  1.13/candidate:   ↑
  1.13/beta:        ↑
  1.13/edge:        ↑
  1.12/stable:      v1.12.9         2019-06-06  (612) 259MB classic
  1.12/candidate:   ↑
  1.12/beta:        ↑
  1.12/edge:        ↑
  1.11/stable:      v1.11.10        2019-05-10  (557) 258MB classic
  1.11/candidate:   ↑
  1.11/beta:        ↑
  1.11/edge:        ↑
  1.10/stable:      v1.10.13        2019-04-22  (546) 222MB classic
  1.10/candidate:   ↑
  1.10/beta:        ↑
  1.10/edge:        ↑
installed:          v1.20.9                    (2361) 221MB classic

**Comment by receperdogan** [comment]
> New snap revisions (2361 and 2362) should be on 1.20/edge, 1.20/beta and 1.20/candidate. This revision is now going through testing to be released on the 1.20/stable channel.
> 
> As @balchua mentioned, this issue was patched a few weeks ago in all tracks (1.19, 1.20, 1.21 and latest). Our CI did not pick up the change on the 1.20 branch. I cannot offer more insight as to why this is the case, I need to do some investigation.
> 
> Apologies for the inconvenience.

Thanks for your quick response. it seems my problem solved too.

**Comment by ktsakalozos** [comment]
A new snap revision is released.
